services/roomService/src/lib/CoveyTownController.ts

```diff
@@ -154,18 +165,48 @@ export default class CoveyTownController {
       return d <= Math.max(0, incomingMessage.broadcastRadius);
     };
     const nearbyPlayerSessions = this._sessions.filter(s => isInChatRadius(s.player));
-    const receivingPlayers = nearbyPlayerSessions.map(s => s.player).filter(p => p.id !== sender.id);
+
+    const notBlocked = (s: PlayerSession) => {
+      const length = this._blockedPlayerSessions.get(session)?.length;
+      for(let i=0; i<(length as number); i++) {
+        if (this._blockedPlayerSessions.get(session)?.includes(s)) {
+          return false;
+        }
+      }
+      return true;
+    };
+    const nearbyUnblockedPlayerSessions = nearbyPlayerSessions.filter(s => notBlocked(s));
```

Comment:
Here's a concise way to do this:
`const nearbyUnblockedPlayerSessions = nearbyPlayerSessions.filter(s => !this._blockedPlayerSessions.get(session)?.includes(s));`